src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
```suggestion
  const women = people.filter(person => ( withChildren
    ? person.sex === 'f' && people.find((child) => child.mother === person.name))
    : person.sex === 'f');
```
Don't duplicate code and use more readable variable names